### DISCUSSION TITLE
Release 3.0

### DISCUSSION THREAD

**Discussion by revoxhere** [revoxhere/duino-coin#1249] [General]
#### New features
* New Arduino code optimizations by @revoxhere in https://github.com/revoxhere/duino-coin/pull/1236 (**Up to 20% more hashrate**) :sunglasses: 
* Added ESP8266 dashboard and code optimizations by @revoxhere in https://github.com/revoxhere/duino-coin/pull/1242 (**Self-hosted dashboard** and small improvements) :sunglasses: 
* Added ESP32 dashboard and code optimizations by @revoxhere in https://github.com/revoxhere/duino-coin/pull/1216 (**Self-hosted dashboard** and small improvements) :sunglasses: 
* New fasthash optimizations for PC Miner by @revoxhere and @hgecode in https://github.com/revoxhere/duino-coin/pull/1243 (**Up to 10 times more hashrate!!!**) :sunglasses: 
#### New languages
* Updated the russian README by @KravchaDev in https://github.com/revoxhere/duino-coin/pull/1110
* Updated the spanish README by @LDarki in https://github.com/revoxhere/duino-coin/pull/1237
* Added a new README language in https://github.com/revoxhere/duino-coin/pull/1113
* README.md fix by @hotmoist in https://github.com/revoxhere/duino-coin/pull/1139
* Added a new language (ukrainian) by @osmolenko in https://github.com/revoxhere/duino-coin/pull/1222
* Added a new language (slovak) in https://github.com/revoxhere/duino-coin/pull/1119
* Added a new language (korean) by @hotmoist in https://github.com/revoxhere/duino-coin/pull/1135
#### Bugfixes and improvements
* /getPool timeout for PC Miner by @revoxhere in https://github.com/revoxhere/duino-coin/pull/1141
* /getPool timeout for AVR miner by @revoxhere in https://github.com/revoxhere/duino-coin/pull/1142
* Add double quote for utf-8 by @TypeNaN in https://github.com/revoxhere/duino-coin/pull/1143
* Added missing translation strings by @revoxhere in https://github.com/revoxhere/duino-coin/pull/1175
* Added block found report and fixed the share rate by @JK-Rolling in https://github.com/revoxhere/duino-coin/pull/1176
* Added a catch to set_title exception when running bash as a background process by @tyrkelko in https://github.com/revoxhere/duino-coin/pull/1179
* Fixed retry_count being always 0 by @TypeNaN in https://github.com/revoxhere/duino-coin/pull/1185
* Fixed the Discord RPC by @Omicron166 in https://github.com/revoxhere/duino-coin/pull/1203
* Small AVR_Miner improvements and bugfixes by @revoxhere in https://github.com/revoxhere/duino-coin/pull/1244
* Fixed AVR Miner feedback bug from 3.0rc
* Fixed AVR Miner board reconnection bug from 3.0rc
* Fixed fasthash accelerations on Windows and Linux binaries from 3.0rc
* Added some missing translation strings

### New Duino contributors - welcome! :blush:
* @KravchaDev made their first contribution in https://github.com/revoxhere/duino-coin/pull/1110
* @TypeNaN made their first contribution in https://github.com/revoxhere/duino-coin/pull/1143
* @JK-Rolling made their first contribution in https://github.com/revoxhere/duino-coin/pull/1176
* @tyrkelko made their first contribution in https://github.com/revoxhere/duino-coin/pull/1179
* @Omicron166 made their first contribution in https://github.com/revoxhere/duino-coin/pull/1203
* @osmolenko made their first contribution in https://github.com/revoxhere/duino-coin/pull/1222

#### ⚠️ NOTICE: This is a major update. Miners using previous versions (2.X) will stop being accepted by the nodes after two weeks of this release getting published.

**Full Changelog**: https://github.com/revoxhere/duino-coin/compare/2.7.4...3.0
As always, please report any encountered issues. Happy mining!

<hr /><em>This discussion was created from the release <a href='https://github.com/revoxhere/duino-coin/releases/tag/3.0'>Release 3.0</a>.</em>

**Comment by techmicrobe** [comment]
How do we upgrade to 3.0 from the current version?

**Comment by Pilske63** [comment]
Does anyone also have problems with the Wifi connection.
Many times  my 5x rig miner disconnect because Wifi is <90

I think this is related to my Ziggo signal strength. I am working with Deco m4 devices. vai speedtest.net i have 150 mbps download / 30mbps upload.

Ik can see via inSIDDer that channel 10 en 44 are crowded in my neighbourhood and signal strength clatters between 25 - 72

Somebody who can help me?
I already made a change in this line ( 90 --> 10)

![image](https://user-images.githubusercontent.com/97939409/149888216-92a83b29-d3a6-41bc-a09a-d9ddc70f7131.png)

**Reply by CosmicShreyas** [reply]
hmmm not sure why

**Comment by Einariux** [comment]
ESP826-12E
GOOD share #26 (28455) hashrate: 9.15 kH/s (3.11s) old 2.75
GOOD share #3 (1535) hashrate: 5.39 kH/s (0.28s)new 3.0
same on Raspberry Pi 3+ 2.7.5 hashrate 1.0mh/s now 524kh/s.

**Reply by rtecheEmbedded** [reply]
on version 3.0 esp12 hasrate drop 50%...

**Comment by gdsmit1** [comment]
ESP32 went from ~32 kH/s  to ~42 kH/s
SparkFun Thing went from ~26 kH/s to ~42 kH/s

But how do you access the ESP32 dashboards?

**Reply by sunsunny83** [reply]
> ESP32 went from ~32 kH/s  to ~42 kH/s
> SparkFun Thing went from ~26 kH/s to ~42 kH/s
> 
> But how do you access the ESP32 dashboards?

Assign (reserve) Static IP to each ESP-32 Miners in WiFi router then directly access those IP in browser after upgrading to 3.0

**Comment by Alfred1023** [comment]
Coding for andriod user, we dont have pc, laptop, computer

**Comment by JACK7JACK** [comment]
How to code nodemcu using android please help me

**Reply by CosmicShreyas** [reply]
u cant u need libraries and board drivers and system drivers and stuffs that cant run on aarch (androids)

**Reply by JACK7JACK** [reply]
I'm still having trouble downloading these 3 libraries
(ESP8266mDNS.h, ESP8266HTTPClient.h, ESP8266WiFi.h)

**Reply by JACK7JACK** [reply]
![Uploading IMG_20220417_154154.jpg…]()

**Comment by Tambahrejo** [comment]
Q coba upgrade ke v.3 tp gagal trs "akun tidak dikenal"

**Comment by Raabitco** [comment]
Dear Rev, the server looks down until now, I cannot do mining for more than 5 days so far, will you update us on the wallet website pleas? do we need to do anything to activate our miners again?

**Comment by johnnybegoode77** [comment]
Windows won't let me install Duino-Coin_3.0_windows.zip
Windows Defender found threats on the Windows Zip file.
Program:Win32/Uwamson.A!ml

"
Uwamson Description. Uwamson is an extremely threatening malware classified as a Trojan. Trojans can perform a multitude of invasive and nefarious actions on the infected systems, depending on the specific goals of the attackers."

**Comment by Tambahrejo** [comment]
when you disable windows defender, how to analyze the virus on pc

**Comment by Tambahrejo** [comment]
I access duino-faucet.com, but a proxy user is detected, even though I
don't activate the VPN, so what's wrong?

**Comment by gibethub** [comment]
PC (DUCO-S1) with Rigname: RaspberryPi2b and Softwarename: Official PC Miner 3.0 with Hashrate: 310265H/s is mining too fast (thats not better). How is it possible that this is error? It is only a Raspberry Pi2b+

**Comment by Jaime-grx** [comment]
There isn't a version for mac. What can I do?

**Reply by LDarki** [reply]
Hi, you can use the src code .py file.

**Reply by Jaime-grx** [reply]
What is that. I want to mine with arduino.

**Reply by LDarki** [reply]
You can use the python file of the AVR_Miner.
There is no a version for mac.

**Reply by joanmauri98** [reply]
Therefore no option for mac users? but I had a previous version that worked

**Comment by sys-256** [comment]
> New Arduino code optimizations by @revoxhere in https://github.com/revoxhere/duino-coin/pull/1236 (**Up to 20% more hashrate**) 😎

According to [this](https://github.com/revoxhere/duco-webservices/commit/1091ab999f94383e080d55d319335bcc36e73d35#commitcomment-56713192), does this mean you're intentionally lowering rewards? Or did you change Kolka in the meanwhile which did go unnoticed to me? If so, I'm still ready to merge my js-sha1 fork with higher hashrate.

**Comment by mihail-soldat** [comment]
Ардуино Уно постоянно отклчается от порт
![Снимок](https://user-images.githubusercontent.com/72151512/157915063-b6e925b4-90df-4fe6-8fa6-de8d2906b39b.JPG)
а

**Comment by vladimir405** [comment]
Здравствуйте. При запуске программы AVR_Miner.exe выходит такая ошибка, у меня Windows 7 32-разрядная. 
Как можно это решить?
![Uploading image0.jpeg…]()

**Comment by ghost** [comment]
I got a error in line 38 pls help me to fix 
Problem ESP8266_Code.ino:38:17: error: 'experimental' has not been declared
